server-common/src/main/java/org/apache/kafka/server/common/TopicIdPartition.java

```diff
@@ -14,43 +14,62 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+package org.apache.kafka.server.common;
 
-package org.apache.kafka.controller;
-
-import java.util.Objects;
 import org.apache.kafka.common.Uuid;
 
-final class TopicIdPartition {
+/**
+ * Represents a partition using its unique topic Id and partition number.
+ */
+public final class TopicIdPartition {
     private final Uuid topicId;
-    private final int partitionId;
+    private final Integer partitionId;
```

Comment:
Why changing to an Integer? An int seems more appropriate.

Reply:
Couldn't use equals and hashCode methods on int

Reply:
We should keep an int here. It is possible to implement equals and hashCode with it (as it was before).